src/console/controllers/MigrateController.php

```diff
@@ -211,8 +210,12 @@ public function beforeAction($action): bool
             $projectConfig->regenerateExternalConfig();
         }
 
-        if (!$this->traitBeforeAction($action)) {
-            return false;
+        try {
+            if (!$this->traitBeforeAction($action)) {
+                return false;
+            }
+        } catch (InvalidConfigException $e) {
+            return true;
         }
```

Comment:
`$this->traitBeforeAction` calls `parent::beforeAction`, which throws `InvalidConfigException` if it can't create the dir.

Technically it will throw `yii\base\Exeption` if this is the `create` action, but I don't think we care. If you can't create the dir, you obviously wont be able to create a migration file.

Reply:
If we are catching the wrong exception, why `try/catch` it in the first place?

Reply:
Not the wrong exception, just some exception silliness here from Yii:

When `migrations` can't be created:

- `migrate/create` will throw `yii\base\Exception`: we're not bothering catching this, because we can't create a migration on a RO fs.
- `migration/*` (anything besides `create`) will throw `InvalidConfigException`, which we are catching